core/src/View/MDSpan/Kokkos_MDSpan_Accessor.hpp

```diff
@@ -212,6 +212,334 @@ struct AtomicAccessorRelaxed {
   }
 };
 
+//=====================================================================
+//============= Reference Counted Accessor and DataHandle =============
+//=====================================================================
+
+template <class ElementType, class MemorySpace>
+class ReferenceCountedDataHandle {
+ public:
+  using value_type   = ElementType;
+  using pointer      = value_type*;
+  using reference    = value_type&;
+  using memory_space = MemorySpace;
+
+  KOKKOS_DEFAULTED_FUNCTION
+  ReferenceCountedDataHandle() = default;
+  explicit ReferenceCountedDataHandle(SharedAllocationRecord<void, void>* rec) {
+    m_tracker.assign_allocated_record_to_uninitialized(rec);
+    m_handle = static_cast<pointer>(get_record()->data());
+  }
+
+  KOKKOS_FUNCTION
+  ReferenceCountedDataHandle(const SharedAllocationTracker& tracker,
+                             pointer data_handle)
+      : m_tracker(tracker), m_handle(data_handle) {}
+
+  // unmanaged ctor
+  template <class OtherElementType,
+            class = std::enable_if_t<std::is_convertible_v<
+                OtherElementType (*)[], value_type (*)[]>>>
+  KOKKOS_FUNCTION ReferenceCountedDataHandle(OtherElementType* ptr)
+      : m_tracker(), m_handle(ptr) {}
+
+  // subview ctor
+  template <class OtherElementType,
```

Comment:
Is *subview* really appropriate ?

Reply:
In some ways yes, in some ways no. Since we don't publicly expose this class it's generally meant as an implementation detail of View (hence subview). But mechanically it can and is used inside submdspan. So I could change that comment.

Reply:
I will update the description of this guy and everything else a bit better (i.e. make more comments)